Tests/FluentAssertions.Json.Shared.Specs/JTokenAssertionsSpecs.cs

```diff
@@ -662,6 +662,56 @@ public void When_jtoken_does_have_element_NotHaveElement_should_fail()
 
         #endregion (Not)HaveElement
 
+        #region HaveElementAndValue
+
+        [Fact]
+        public void When_jtoken_has_element_and_value_HaveElementAndValue_should_succeed()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var subject = JToken.Parse("{ 'id': 42 }");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act & Assert
+            //-----------------------------------------------------------------------------------------------------------
+            subject.Should().HaveElementAndValue("id", "42");
```

Comment:
What if any of the two arguments are empty or `null`?

Reply:
I guess I'll follow the same rules as `HaveElement` and `HaveValue`. Any better ideas?

Reply:
Don't count on the existing methods to deal with this properly. Let's just make these correct.

Reply:
I see. What's our agenda then?

Reply:
None. I want to make sure new code is properly validating all edge cases.

Reply:
Right, so what should the expected behavior be on nulls?

Reply:
I guess if the `subject` is `null`, it should throw the usual assertion failure. But if either of the parameters are `null`, I would throw an `ArgumentNullException`. If the first parameter is empty, it should throw an `ArgumentException`. I'm in doubt about the second one, but I guess expecting an empty value is something to be expected. What do you think>

Reply:
Yep. Makes perfect sense to me.

Reply:
I think the name parameter makes perfect sense to validate both scenarios, unlike the value parameter which should not be validated, because it makes sense for it to be verified as `null` too.

Reply:
I added some tests to cover these scenarios (https://github.com/fluentassertions/fluentassertions.json/pull/40/commits/50cef9e51c2686628ddd448d7c93b47e2f0a508a):

- `ArgumentNullException` if `expectedElement` is `null`.
- `ArgumentException` if `expectedElement` is empty or whitespace only.
- Verify assertion succeeds if expected and actual JSON values are `null`.


Due to #41 I can't actually run the tests and can only hope for CI to give me the results.